### PR TITLE
Fix unintentional UB in SIMD tests

### PR DESCRIPTION
**Pull request by saethlin**
r? @workingjubilee

### CODE REVIEW

**Review by calebzulawski — APPROVED** [review]

File: tests/ui/simd/issue-89193.rs [hunk 1]
```diff
@@ -17,13 +17,13 @@ extern "platform-intrinsic" {
 fn main() {
     let x: [usize; 4] = [10, 11, 12, 13];
     let default = x4(0_usize, 1, 2, 3);
-    let mask = x4(1_i32, 1, 1, 1);
+    let mask = x4(u8::MAX, u8::MAX, u8::MAX, u8::MAX);
```
Comment (workingjubilee) [inline tests/ui/simd/issue-89193.rs:20]: This is actually considered a signed integer type for... some reason, I don't know if it actually matters.

```suggestion
    let all_set = u8::MAX as i8; // aka -1
    let mask = x4(all_set, all_set, all_set, all_set);
```

Comment (calebzulawski) [reply tests/ui/simd/issue-89193.rs]: I think `repr(simd)` is okay with either, but we tend to use signed.

Comment (saethlin) [reply tests/ui/simd/issue-89193.rs]: unsigned doesn't compile (I force-pushed this away): https://github.com/rust-lang/rust/actions/runs/4158968893/jobs/7194581100

**Comment by workingjubilee** [comment]
actually should make sure.
@bors r+ rollup

**Pull request merged**